### ISSUE TITLE
Autosplit function and files without annotations

### ISSUE THREAD

**Issue by kinoute** [ultralytics/yolov5#2228]
## ❔Question

The `autosplit()` function seems handy to create a train/validation set from custom data stored in different folders. But it seems that it basically adds everything it finds in these folders (i.e. images). It doesn't check if there is an annotation file associated to a file before appending it to the autosplit* files.

From what I know and what I can read in the WIKI, an annotation file is not needed in case there is nothing to detect inside an image. So my question is: should this function _only_ use images with annotation files associated to them to create the train/validation set? From my understanding, the images without annotations are ignored when we start a training. Or am I missing something?

Thanks!

**Comment by glenn-jocher** [comment]
@kinoute autosplit() splits images into groups suitable for train/val/test splits. 

Labels are a separate topic. All images in a dataset are used for training. If a corresponding label file is found, those labels are used for training. COCO has 1021 images with no label files for example.

**Comment by kinoute** [comment]
I see. It makes the creation of dataset very difficult since the absence of an annotation file could mean two things: either there is nothing to detect inside the image, either the image wasn't labelled yet. But it's nice to have the confirmation that yolov5 treats also images without annotations files.

Now I need to find a way to track what I annotated and what I didn't, before creating a "yolo" dataset version.

**Comment by glenn-jocher** [comment]
@kinoute non-annotated image training is a YOLOv5 feature but not a requirement. If it helps your personal workflow empty annotations files are treated identically to missing annotations files by the dataloader. i.e. a image.txt file with nothing in it also works well.

**Comment by kinoute** [comment]
@glenn-jocher Thanks! That seems to be a nice alternative. Is training yolo only with images annotated/with objects in it a really bad idea? That would obviously make things way easier when creating the dataset.

**Comment by glenn-jocher** [comment]
@kinoute it depends on your priorities and your results. If you prioritize increasing Precision at the expense of Recall, and if you are observing excess FPs after training then you will want to add extra background images. The workflow for this if your data.yaml points to a directory, as in coco128.yaml, is simply to 1) copy new images to the training images directory, and 2) retrain.
